lib/unifex/code_generator.ex

```diff
@@ -2,55 +2,58 @@ defmodule Unifex.CodeGenerator do
   @moduledoc """
   Behaviour for code generation.
   """
+
   alias Unifex.Specs
 
   @type code_t :: String.t()
+  @type generated_code_t :: {header :: code_t, source :: code_t, generator :: module()}
 
   @callback identification_constant() :: String.t()
+  @callback subdirectory() :: String.t()
```

Comment:
why is using just module name insufficient?

Reply:
because module name is mixed upper/lower case while we had directory names lowercase. I went for a callback because I think it's not elegant to enforce simply lowercasing module name suffix

I think on adding a default impl for this function though

Reply:
To me, it seems not elegant to couple generating code stuff with IO-related stuff. Maybe naming it more universally (`io_name` ?) would help?

Reply:
Well don't code generation involve setting up proper directory structure? Especially as TieHeader **code generator** uses this field.

`io_name` seems more vague to me than `subdirectory`

Reply:
Code generators rely on the directory structure whenever they have to, but they don't specify it. Nevertheless, I would at least name that callback in a way suggesting either that it's _the_ directory where the code lands or that it's a version of the name of generator used to do IO operations.